### PR TITLE
build: add "CPU_NUMS" to improve compilation speed.

### PR DESCRIPTION
**Pull request by darlentar**

### CODE REVIEW

File: Makefile.in [hunk 1]
```diff
@@ -530,12 +530,13 @@ submodules/orc/NOTICE: .gitmodules
...
 src/orc/wrappers.o: $(BUNDLE_DIR)/lib/liborc.a
 
+# Seems we cannot build orc in parallel
 $(BUNDLE_DIR)/lib/liborc.a: $(BUNDLE_DIR)/date submodules/orc/NOTICE
 	@echo 'Building liborc'
 	@mkdir -p submodules/orc/build
 	@cd submodules/orc/build &&\
 	 CPPFLAGS='$(CFLAGS) $(CPPFLAGS)' cmake .. -DBUILD_JAVA=OFF -DCMAKE_BUILD_TYPE=DEBUG -DCMAKE_INSTALL_PREFIX:PATH=$(PWD)/$(BUNDLE_DIR) &&\
-	 make install
+	 make -j@CPU_NUMS@ install
```
Comment (rixed) [inline Makefile.in]: Nope, but $(MAKE) would be OK if it works.

File: Makefile.in [hunk 2]
```diff
@@ -530,12 +530,13 @@ submodules/orc/NOTICE: .gitmodules
 # We need the ORC include files to be installed for that one:
 src/orc/wrappers.o: $(BUNDLE_DIR)/lib/liborc.a
 
+# Seems we cannot build orc in parallel
```
Comment (rixed) [inline Makefile.in]: If that's true, then why bother adding `-j` ?

File: configure [hunk 3]
```diff
@@ -573,6 +573,7 @@ cross_compiling=no
 subdirs=
 MFLAGS=
 MAKEFLAGS=
+CPU_NUMS=`grep -c '^processor' /proc/cpuinfo`
```
Comment (rixed) [inline configure]: That's supposed to be a user setting (`make -j` should be enough and not decided by configure).

File: docker/builder.in [hunk 4]
```diff
@@ -30,7 +30,7 @@ check() {
 
   checkout "$rev"
   build
-  make check
+  make -j@CPU_NUMS@ check
```
Comment (rixed) [inline docker/builder.in]: Nope. This patch capture how many CPU you have on your desktop, and write it in a docker file that's then run on other machines.
Let's rather have a cpu-num option to builder (again, that's a user parameter), if we really have to.

What problem is this patch trying to solve?

File: Makefile.in (hunk 1 above)
Comment (darlentar) [reply Makefile.in]: You right

File: Makefile.in (hunk 2 above)
Comment (darlentar) [reply Makefile.in]: For consistency

File: configure (hunk 3 above)
Comment (darlentar) [reply configure]: Ok

File: docker/builder.in (hunk 4 above)
Comment (darlentar) [reply docker/builder.in]: What am I trying to solve is written in the first message: improve build speed.
What do you mean by "run on other machines" ? It's a dev file, for me it's ran on dev machines.

**Pull request closed without merging**